src/main/java/org/openstreetmap/atlas/geography/atlas/items/Relation.java

```diff
@@ -93,6 +99,44 @@ public String configurableString(final String betweenEachMemberAndRelation,
         return builder.toString();
     }
 
+    /**
+     * "Flattens" the relation by returning the set of non-Relation members. Adds any non-Relation
+     * members to the set, then loops on any Relation members to add their non-Relation members as
+     * well. Keeps track of Relations whose identifiers have already been operated on, so that
+     * recursively defined relations don't cause problems.
+     *
+     * @return a Set of AtlasObjects all related to this Relation, with no Relations.
+     */
+    public Set<AtlasObject> flatten()
+    {
+        final Set<AtlasObject> relationMembers = new HashSet<>();
+        final Deque<AtlasObject> toProcess = new LinkedList<>();
+        final Set<Long> relationsSeen = new HashSet<>();
+        AtlasObject polledMember;
+
+        toProcess.add(this);
+        while (!toProcess.isEmpty())
+        {
+            polledMember = toProcess.poll();
+            if (relationsSeen.contains(polledMember.getIdentifier()))
```

Comment:
This would be a really rare case, but OSM identifiers are only unique within their type - meaning a Relation and Way can have the same identifier, but two Relations cannot. Because of this, it's technically possible for two different `AtlasEntity`s to have the same identifier. To avoid this, we also want to check the type. I would add an explicit check that the `polledMember` is a `Relation`, in addition to having been seen. You can add this check in at line 127 when you're checking for Relations.

Reply:
I'm probably reading it wrong - won't the issue still be present? If we have a `polledMember`, that's a `Node`, with the same identifier as an already seen `Relation` - that `Node` wouldn't make it into the final return set, right? May be worth adding a unit test case for this too :)

Reply:
oh, I see, this is my mistake-- I had initially thought you were just referring to the adding of an ID as seen. but, obviously, it should apply to the check as well. I'll move it in, and also make a unit test.